TheMET/TheMET/Network/NetworkManager.swift

```diff
@@ -10,40 +10,61 @@ import Foundation
-        guard var components = URLComponents(string: urlString) else {
-            completion(nil)
-            return
-        }
-        components.queryItems = parameters.map { (key, value) in
-            URLQueryItem(name: key, value: value)
-        }
-        components.percentEncodedQuery = components.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%2B")
-        guard let url = components.url else {
-            completion(nil)
-            return
-        }
-        var urlRequest = URLRequest(url: url)
-        urlRequest.httpMethod = "GET"
-        let urlSession = URLSession.shared
-        let fileDownloadTask = urlSession.dataTask(
-            with: urlRequest
-        ) { data, _, error in
-            let executeCompletionOnMain: ([String: Any]?) -> Void = { loadedData in
-                DispatchQueue.main.async {
-                    completion(loadedData)
-                }
+        var lastRequestTime = Date()
+            guard var components = URLComponents(string: urlString) else {
+                completion(nil)
+                return
+            }
+            components.queryItems = parameters.map { (key, value) in
+                URLQueryItem(name: key, value: value)
             }
-            guard error == nil,
-                  let data = data else {
-                executeCompletionOnMain(nil)
+            components.percentEncodedQuery = components.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%2B")
+            guard let url = components.url else {
+                completion(nil)
                 return
             }
-            let result = try? JSONSerialization.jsonObject(with: data) as? [String: Any]
-            executeCompletionOnMain(result)
+            var urlRequest = URLRequest(url: url)
+            urlRequest.httpMethod = "GET"
+            let urlSession = URLSession.shared
+            let fileDownloadTask = urlSession.dataTask(
+                with: urlRequest
+            ) { data, _, error in
+                let executeCompletionOnMain: ([String: Any]?) -> Void = { loadedData in
+                    DispatchQueue.main.async {
+                        completion(loadedData)
+                    }
+                }
+                guard error == nil,
+                      let data = data else {
+                    executeCompletionOnMain(nil)
+                    return
+                }
+                let result = try? JSONSerialization.jsonObject(with: data) as? [String: Any]
+                executeCompletionOnMain(result)
+            }
+        while !self.timeIntervalBWLastRequestTimeToNow(lastRequest: lastRequestTime) {
+            DispatchQueue.global().asyncAfter(deadline: DispatchTime.now() + .milliseconds(Int(12.5)), qos: .background) {
+            }
         }
         fileDownloadTask.resume()
+        lastRequestTime = .now
     }
     
+   func timeIntervalBWLastRequestTimeToNow(lastRequest: Date) -> Bool {
```

Comment:
1) This method is supposed to be `private`
2) Naming. This method is named like it should return a time between now and last-request date. Actually it returns a `Bool` which is true if this time is more than 12.5ms. Please change the name to smth prefixed with `is`